### PR TITLE
Add section about changing PHP versions

### PR DESCRIPTION
**Pull request by fritzmg**
See #973

### CODE REVIEW

**Review by Toflar — CHANGES_REQUESTED** [review]
Can we have a reason as to why this is required? Just a simple sentence about the fact that platform requirements are also considered in Composer? Otherwise you only know that you have to do it but you don't understand why.

**Comment by fritzmg** [comment]
Well, it already says

> This ensures compatibility of your installed packages with the respective PHP version.

but I'd expand it then to

> This ensures compatibility of your installed packages with the respective PHP version, since each package (including Contao itself, installed Contao extensions and other third-party packages) can require a specific PHP version that it is known to be compatible with.

wdyt?

**Comment by Toflar** [comment]
👍 Maybe also "and certain PHP extensions".

File: docs/manual/installation/system-requirements.de.md [hunk 1]
```diff
@@ -126,7 +126,9 @@ PHP **7.2.0** erforderlich.
 
 Falls die PHP Version einer laufenden Contao Instanz geändert werden soll, sollte immer ein volles `composer update` nach dem Wechsel
 durchgeführt werden. Dies ist besonders dann wichtig, wenn zwischen Haupt-Versionen gewechselt wird, z. B. von PHP 7.x zu PHP 8.x - oder
-umgekehrt. Dadurch wird für die installierten Pakete die Kompatibilität zu der jeweiligen PHP Version sichergestellt.
+umgekehrt. Dadurch wird für die installierten Pakete die Kompatibilität zu der jeweiligen PHP Version sichergestellt, da jedes Paket
+(inklusive Contao selbst, installierte Contao Extensions oder Dritt-Pakete) nach spezifischen PHP Versionen und PHP Extensions verlangen
```
Comment (netzarbeiter) [inline docs/manual/installation/system-requirements.de.md]:
```suggestion
(inklusive Contao selbst, installierte Contao-Extensions oder Dritt-Pakete) nach spezifischen PHP-Versionen und PHP-Extensions verlangen
```

**Review by Toflar — APPROVED** [review]

**Review by netzarbeiter — CHANGES_REQUESTED** [review]
Thank you, Fritz

**Review by netzarbeiter — APPROVED** [review]

**Comment by FelixOliverLange** [comment]
I'm really not an expert on PHP nor on composer, so please forgive me my ignorance. But wouldn't running the update be required before the PHP version switch @fritzmg? After all, the dependencies must support the respective version before it is being run, otherwise the dependency incompatibility is noted by composer if you e.g. attempt an update or try to add an unrelated package.

**Comment by fritzmg** [comment]
@FelixOliverLange composer installs the dependencies for which ever PHP version it detects. Thus running a package update before you switch will not change anything (unless you are overriding the PHP version in your `composer.json`). Also updating the dependencies to reflect a newer PHP version change _before_ you actually switch to a new PHP version will break the application even more likely, since you might then get packages that are incompatible with your _current_ PHP version.

**Review by netzarbeiter — APPROVED** [review]
Thank you, Fritz

**Pull request merged**